ecosystem/indexer-grpc/indexer-grpc-table-info/src/internal_indexer_db_service.rs

```diff
@@ -57,18 +76,45 @@ impl InternalIndexerDBService {
         Arc::clone(&self.db_indexer)
     }
 
-    pub async fn run(&mut self) {
-        let mut start_version = self
+    pub async fn get_start_version(&self, node_config: &NodeConfig) -> Result<Version> {
+        let indexer_version = self.db_indexer.indexer_db.get_persisted_version()?;
+        let fast_sync_enabled = node_config
+            .state_sync
+            .state_sync_driver
+            .bootstrapping_mode
+            .is_fast_sync();
+        let mut db_min_version = self.db_indexer.get_main_db_lowest_viable_version()?;
+
+        // Wait till fast sync is done
+        while fast_sync_enabled && db_min_version == 0 {
```

Comment:
in theory fast sync can sync to version 0.. on devnet, for example. 

I thought we talked about explicitly writing down something when fast sync is done?

Reply:
okay I see you are reading that a few lines down. Shall we read that in a loop?